### PR TITLE
[chore] update golangci to v1.55.2

### PR DESCRIPTION
**Pull request by faddat**
## 1. Summary
Update golangci and fix some issues. 

Earlier, this PR used "latest" but it turns out you just need to specify the patch version when choosing a golangci-lint version :).

### CODE REVIEW

**Comment by faddat** [comment]
@joe-bowman I think this is good to go but can you give it a once-over? don't want to ninja merge this since it has actual changes.

**Review by joe-bowman — APPROVED** [review]
ship it

**Pull request merged**
